src/components/sbb-file-selector/sbb-file-selector.tsx

```diff
@@ -0,0 +1,293 @@
+
+  /** Whether the component is disabled. */
+  @Prop() public disabled: boolean;
+
+  /** The list of selected files. */
+  @State() private _files: File[];
+
+  /** Current document language used for translations. */
+  @State() private _currentLanguage = documentLanguage();
+
+  /**
+   * State of listed named slots, by indicating whether any element for a named slot is defined.
+   */
+  @State() private _namedSlots = createNamedSlotState('error');
+
+  /** An event which is emitted each time the file list changes. */
+  @Event({
+    eventName: 'file-changed',
+  })
+  public fileChangedEvent: EventEmitter<File[]>;
+
+  /** An event emitted on error. */
+  @Event({ bubbles: true }) public error: EventEmitter;
+
+  @Element() private _element!: HTMLElement;
+
+  private _loadButton: HTMLElement;
+  private _dragTarget: HTMLElement;
+  private _suffixes: string[] = ['B', 'kB', 'MB', 'GB', 'TB'];
+
+  private _handlerRepository = new HandlerRepository(
+    this._element,
+    languageChangeHandlerAspect((l) => (this._currentLanguage = l)),
+    namedSlotChangeHandlerAspect((m) => (this._namedSlots = m(this._namedSlots))),
+  );
+
+  public connectedCallback(): void {
+    this._handlerRepository.connect();
+  }
+
+  public disconnectedCallback(): void {
+    this._handlerRepository.disconnect();
+  }
+
+  private _blockEvent(event): void {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
+  private _checkFileEquality(file1: File, file2: File): boolean {
+    return (
+      file1.name === file2.name &&
+      file1.size === file2.size &&
+      file1.lastModified === file2.lastModified
+    );
+  }
+
+  private _onDragEnter(event): void {
+    if (!this.disabled) {
+      this._setDragState(event.target, true);
+      this._blockEvent(event);
+    }
+  }
+
+  private _onDragLeave(event): void {
+    if (!this.disabled && event.target === this._dragTarget) {
+      this._setDragState();
+      this._blockEvent(event);
+    }
+  }
+
+  private _onFileDrop(event): void {
+    if (!this.disabled) {
+      this._setDragState();
+      this._blockEvent(event);
+      try {
+        const files = this.multiple ? event.dataTransfer.files : [event.dataTransfer.files[0]];
+        this._createFilesList(files);
+      } catch (e) {
+        this.error.emit(e);
+      }
+    }
+  }
+
+  private _onFocus(): void {
+    if (sbbInputModalityDetector.mostRecentModality === 'keyboard') {
+      toggleDatasetEntry(this._loadButton, 'focusVisible', true);
+    }
+  }
+
+  private _onBlur(): void {
+    if (sbbInputModalityDetector.mostRecentModality === 'keyboard') {
+      toggleDatasetEntry(this._loadButton, 'focusVisible', false);
+    }
+  }
+
+  private _setDragState(dragTarget = undefined, isDragEnter = false): void {
+    this._dragTarget = dragTarget;
+    toggleDatasetEntry(this._element, 'active', isDragEnter);
+    toggleDatasetEntry(this._loadButton, 'active', isDragEnter);
+  }
+
+  private _readFiles(event): void {
```

Comment:
Please add types

Reply:
The event type here should be a DOM `Event`, which conflicts with the Stencil `Event` type; as in other points, i left it untyped..

Reply:
added DomEvent type